schemas/nl.basjes/yauaa_context/jsonschema/1-0-4

```diff
@@ -0,0 +1,234 @@
+{
+    "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+    "description": "Schema for an entity context generated by the YAUAA enrichment after parsing the user agent",
+    "self": {
+        "vendor": "nl.basjes",
+        "name": "yauaa_context",
+        "format": "jsonschema",
+        "version": "1-0-4"
+    },
+    "type": "object",
+    "properties": {
+        "deviceClass": {
+            "description": "See https://yauaa.basjes.nl/README-Output.html",
+            "enum": ["Desktop", "Anonymized", "Unknown", "UNKNOWN", "Mobile", "Tablet", "Phone", "Watch", "Virtual Reality", "eReader", "Set-top box", "TV", "Game Console", "Home Appliance", "Handheld Game Console", "Voice", "Car", "Robot", "Robot Mobile", "Spy", "Hacker", "Augmented Reality", "Robot Imitator"]
+        },
+        "deviceName": {
+            "description": "Example: Google Nexus 6",
+            "type": ["string", "null"],
```

Comment:
Why the change from type string to type `[string, null]`?  I'm not against it if you have a good reason.  It struck me as odd because enrich doesn't output nulls.

Reply:
They are optional fields, so presumably at some point they may be sent with a `null` value. We even have it as a check in igluctl. 

I appreciate we currently do not do that in enrich, but if we change it in the future, we won't need to worry about the schema not allowing it. It's part of trying to make the schema more future-proof.

Reply:
My gut feeling at the moment is like this:

If you're going to allow nulls for some fields, then you need to allow nulls for all fields, or else it's not consistent between fields.  Which means allow nulls for the enum fields too, like `operatingSystemClass`.  Which means add `null` to the enum array.

If you're adding `null` to the enum array then you need to be very very sure that RDB Loader will not choke on the schema migration.  Remember that RDB Loader *immediately* (and probably wrongly) tries to migrate the table as soon as the schema is published; it does not wait until it sees the first `1-0-4` event.  Remember also, we need to be sure that old versions of RDB Loader also don't choke on the schema migration, otherwise we might break oss pipelines just by publishing a schema.

Now RDB Loader _probably_ handles this OK.  I'm just saying we need to be 100% sure before merging this.

For that reason.... personally I would avoid adding the nulls, and put it back to be more like how it was in 1-0-3.  There is no reason whatsoever to add nulls to this schema.  I see no reason why enrich will ever change in future and to start emitting nulls.

Yes we want to make the schema future-proof, but I see it as future-proofing against things we are not in control of, like crazy user agent strings.  Whereas nulls are something we are completely in control of, so we don't need to future-proof against nulls.

Reply:
I think that is a sensible idea. I was trying to make igluctl happy, but I agree that in this instance, where we're in control of this, there is no need to worry.